src/Sylius/Behat/Context/Api/Admin/ManagingProductReviewsContext.php

```diff
@@ -0,0 +1,274 @@
+    {
+        $this->applyTransition((string) $productReview->getId(), ProductReviewTransitions::TRANSITION_ACCEPT);
+    }
+
+    /**
+     * @When I reject the :productReview product review
+     */
+    public function iRejectTheProductReview(ReviewInterface $productReview): void
+    {
+        $this->applyTransition((string) $productReview->getId(), ProductReviewTransitions::TRANSITION_REJECT);
+    }
+
+    /**
+     * @Then /^(this product review) status should be "([^"]+)"$/
+     */
+    public function thisProductReviewStatusShouldBe(ReviewInterface $productReview, string $status): void
+    {
+        $this->assertIfReviewHasElementWithValue($productReview, 'status', $status);
+    }
+
+    /**
+     * @Then /^I should be notified that it has been successfully (accepted|rejected)$/
+     */
+    public function iShouldBeNotifiedThatItHasBeenSuccessfullyUpdated(string $action): void
+    {
+        $this->assertIfReviewHasElementWithValue($this->sharedStorage->get('product_review'), 'status', $action);
+    }
+
+    /**
+     * @When I delete the :productReview product review
+     */
+    public function iDeleteTheProductReview(ReviewInterface $productReview): void
+    {
+        $this->sharedStorage->set('product_review_id', $productReview->getId());
+
+        $this->client->delete($this->resource, (string) $productReview->getId());
+    }
+
+    /**
+     * @Then this product review should no longer exist in the registry
+     */
+    public function thisProductReviewShouldNoLongerExistInTheRegistry(): void
+    {
+       $id = (string) $this->sharedStorage->get('product_review_id');
+       Assert::false(
+           $this->isItemOnIndex('id', $id),
+           sprintf('Product review with id %s exist', $id)
+       );
+    }
+
+    /**
+     * @Then I should be notified that :element is required
+     */
+    public function iShouldBeNotifiedThatElementIsRequired(string $element): void
+    {
+        Assert::contains($this->client->getError(), sprintf('%s: Review %s should not be blank', $element, $element));
+    }
+
+    /**
+     * @Then /^(this product review) should still be titled "([^"]+)"$/
+     */
+    public function thisProductReviewTitleShouldBeTitled(ReviewInterface $productReview, string $title): void
+    {
+        $this->assertIfReviewHasElementWithValue($productReview, 'title', $title);
+    }
+
+    /**
+     * @Then /^(this product review) should still have a comment "([^"]+)"$/
+     */
+    public function thisProductReviewShouldStillHaveAComment(ReviewInterface $productReview, string $comment): void
+    {
+        $this->assertIfReviewHasElementWithValue($productReview, 'comment', $comment);
+    }
+
+    /**
+     * @Then /^average rating of (product "[^"]+") should be (\d+)$/
+     */
+    public function thisProductAverageRatingShouldBe(ProductInterface $product, int $averageRating): void
+    {
+        // todo, this step should be in product context which will be covered in separate RP
```

Comment:
after consultation with @Zales0123  I left this step without coverage, this implementation should be in product context which will be configured later

Reply:
IIRC this is the format we try to keep:
```suggestion
        // TODO: this step should be in product context which will be covered in separate RP
```